bump_and_tag.sh

```diff
@@ -0,0 +1,141 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# This script is used for
+# a) creating a new tagged release of go.opentelemetry.io/contrib
+# b) bumping the referenced version of go.opentelemetry.io/otel
+#
+# The options can be used together or individually.
+#
+set -e
+
+help()
+{
+   printf "\n"
+   printf "Usage: %s [-o otel_tag] [-t tag]\n" "$0"
+   printf "\t-o Otel release tag. Update all go.mod to reference go.opentelemetry.io/otel <otel_tag>.\n"
+   printf "\t-t New unreleased tag. Update all go.mod with this tag.\n"
+   exit 1 # Exit script after printing help
+}
+
+while getopts "t:o:" opt
+do
+   case "$opt" in
+      t ) TAG="$OPTARG" ;;
+      o ) OTEL_TAG="$OPTARG" ;;
+      ? ) help ;; # Print help
+   esac
+done
+
+
+validate_tag() {
+    local tag_=$1
+    SEMVER_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(\\-[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?(\\+[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?$"
+    if [[ "${tag_}" =~ ${SEMVER_REGEX} ]]; then
+	    printf "%s is valid semver tag.\n" "${tag_}"
+    else
+	    printf "%s is not a valid semver tag.\n" "${tag_}"
+	    return 1
+    fi
+}
+
+# Print help in case parameters are empty
+if [[ -z "$TAG" && -z "$OTEL_TAG" ]]
+then
+    printf "At least one of '-o' or '-t' must be specified.\n"
+    help
+fi
+
+
+## Validate tags first
+if [ -n "${OTEL_TAG}" ]; then
+    validate_tag "${OTEL_TAG}" || exit $?
+    # check that OTEL_TAG is a currently released tag for go.opentelemetry.io/otel
+    TMPDIR=$(mktemp -d "/tmp/otel-contrib.XXXXXX") || exit 1
+    trap "rm -fr ${TMPDIR}" EXIT
+    (cd "${TMPDIR}" && go mod init tagtest)
+    # requires go 1.14 for support of '-modfile'
+    if ! go get -modfile="${TMPDIR}/go.mod" -d -v "go.opentelemetry.io/otel@${OTEL_TAG}"; then
+        printf "Otel tag %s does not appear to exist.\n" "${OTEL_TAG}"
+        exit 1
+    fi
+fi
+if [ -n "${TAG}" ]; then
+    validate_tag "${TAG}" || exit $?
+    TAG_FOUND=$(git tag --list "${TAG}")
+    if [[ ${TAG_FOUND} = "${TAG}" ]] ; then
+        printf "Tag %s already exists\n" "${TAG}"
+        exit 1
+    fi
+fi
+
+cd $(dirname $0)
+
+if ! git diff --quiet; then \
+    printf "Working tree is not clean, can't proceed\n"
+    git status
+    git diff
+    exit 1
+fi
+
+declare BRANCH_NAME=pre_release_${TAG}
+if [ -z "${TAG}" ]; then
+    BRANCH_NAME=bump_otel_${OTEL_TAG}
+fi
+
+patch_gomods() {
+    local pkg_=$1
+    local tag_=$2
+    # now do the same for all the directories underneath
+    PACKAGE_DIRS=$(find . -mindepth 2 -type f -name 'go.mod' -exec dirname {} \; | egrep -v 'tools' | sed 's|^\.\/||' | sort)
+    # quote any '.' characters in the pkg name
+    local quoted_pkg_=${pkg_//./\\.}
+    for dir in $PACKAGE_DIRS; do
+	    cp "${dir}/go.mod" "${dir}/go.mod.bak"
+	    sed "s|${quoted_pkg_}\([^ ]*\) v[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*[^0-9]*.*$|${pkg_}\1 ${tag_}|" "${dir}/go.mod.bak" >"${dir}/go.mod"
```

Comment:
Should the `SEMVER_REGEX` var get moved to the global scope (and made `readonly`) so it can be used here?

Reply:
Yes